### ISSUE TITLE
Hashtags/Tags Not Maintaining Capitals

### ISSUE THREAD

**Issue by timnolte** [pfefferle/wordpress-activitypub#204]
I was noticing that if people were to follow my WordPress site even though I'm intentionally using uppercase letters in how I'm defining those hashtags for the sake of accessibility the `%hashtags%` template placeholder is just pulling the WordPress stored taxonomy in all lowercase. I haven't had a chance to dig into the code to determine if there is a filter I should be using. Understandably, if people create WordPress Tags with spaces those wouldn't be proper hashtags on Mastodon, though for the sake of promoting proper accessibility I would think that enforcing capital letter for word starts, even removing spaces for a standard would be the right way to go.

> if you go to the plugin settings you can control the output of the ActivityPub posts. Currently I'm using a Custom format as follows:
```
<p><strong>%title%</strong></p>

%excerpt%

<p>%hashtags%</p>

<p>%shortlink%</p>
```
My posts on Mastodon look like this:
![image](https://user-images.githubusercontent.com/4312062/204116657-554dea10-7ae6-47b0-b254-0e7961fe03f3.png)

_Originally posted by @timnolte in https://github.com/pfefferle/wordpress-activitypub/issues/182#issuecomment-1328157085_

**Comment by pfefferle** [comment]
Should be fixed with https://github.com/pfefferle/wordpress-activitypub/pull/184

**Comment by YourAutisticLife** [comment]
(First, thanks for the plugin. I tried writefreely but I quickly found that I had to repost everything it posted to the fediverse... and that's exactly what I'm trying to avoid.)

The problem with hashtags being converted all to lowercase is still present in my own installation. I'm running 0.17.0. For instance, my articles are usually tagged YourAutisticLife in WordPress, but when it goes to the fediverse, I get #yourautisticlife.

The OP says "for the sake of accessibility." The issue here is that people who use screen readers for whatever reason are not happy with hashtags that are made up of many words but that are all lowercase. Their screen reader is unable to make sense of it. I've taken to being careful when I enter tags manually. It would be nice if the plugin would honor the choices I make in WordPress.

Is there some setting I'm ignoring?

**Comment by pfefferle** [comment]
This will change with version 1.0.0. It will keep uppercase and transforms `-`, `_` and `.` to CamelCase. That means:

* `two-words` -> `twoWords`
* `two.words` -> `twoWords`
* `two_words` -> `twoWords`
* `two words` -> `twoWords`
* `Two Words` -> `TwoWords`
* `Two words` -> `TwoWords`
* `TwoWords` -> `TwoWords`

**Comment by YourAutisticLife** [comment]
Great!
